htdocs/contrat/services_list.php

```diff
@@ -467,14 +479,15 @@
 
 print_barre_liste($title, $page, $_SERVER["PHP_SELF"], $param, $sortfield, $sortorder, $massactionbutton, $num, $nbtotalofrecords, 'contract', 0, '', '', $limit);
 
-if ($sall) {
+if (isset($sall)) {
```

Comment:
if ($sall) 
is not similar to 
if (isset($sall)) {
If $sall = '', the isset is true but the if ($all) is true only if value is something not null, not empty, not 0.

Use instead 
if (!empty($sall)) {